### ISSUE TITLE
[FEATURE] MCP: support resources and prompts

### ISSUE THREAD

**Issue by jmartisk** [langchain4j/langchain4j#2304]
see  https://spec.modelcontextprotocol.io/specification/server/prompts/ and https://spec.modelcontextprotocol.io/specification/server/resources/

**Comment by langchain4j** [comment]
@jmartisk did you try to use resources already? Any idea how to integrate that in LC4j?

**Comment by jmartisk** [comment]
No, not yet, and I don't think LC4j has any similar abstraction that maps well onto it, right? So the user would probably just directly use a `McpClient` to obtain representations of resources. Or, potentially we could integrate it into RAG, maybe a content retriever that receives a Query, asks for a list of resources, and then it would probably have to ask an LLM to find out which resources are potentially relevant to the query.

**Comment by langchain4j** [comment]
Yeah, seems like `ContentRetreiver` is the right interface for this. I am not sure it will support all the MCP resources features though

**Comment by sebastienblanc** [comment]
And for the support of the prompts could that map to any LC4j abstraction ?

**Comment by dliubarskyi** [comment]
@sebastienblanc perhaps `PromptTemplate` or [system message provider](https://docs.langchain4j.dev/tutorials/ai-services#system-message-provider)

**Comment by JusticeN** [comment]
The whole mcp server can be build using vertx or spark library. Both are lightweight and easy to use.

All server logic can then be implemented using an 
```
@McpTools(...), @McpRessource(...)
```
method annotation.

To implement a mcp server developer can extend an abstract class (AbstractMCPServer) with preimplantation of base methods like ping, tools and resources list

**Comment by dliubarskyi** [comment]
@JusticeN you might want to check https://quarkus.io/blog/mcp-server/

**Comment by JusticeN** [comment]
Thanks it looks interesting

**Comment by jmartisk** [comment]
@dliubarskyi please reopen this, prompts are still missing :)
